staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/handler.go

```diff
@@ -237,42 +262,40 @@ func (rdm *resourceDiscoveryManager) ServeHTTP(resp http.ResponseWriter, req *ht
 		rdm.lock.RUnlock()
 
 		// Document does not exist, recreate it
-		rdm.lock.Lock()
-		defer rdm.lock.Unlock()
-
-		// Now that we have taken exclusive lock, check to see if another thread
-		// recreated the document while we were waiting for the lock
-		response, etag = rdm.cachedResponse, rdm.cachedResponseETag
-		if response == nil {
-			// Re-order the apiGroups by their insertion order
-			groups := []metav1.APIGroupDiscovery{}
-			for _, v := range rdm.apiGroups {
-				groups = append(groups, *v)
+		func() {
+			rdm.lock.Lock()
+			defer rdm.lock.Unlock()
+
+			// Now that we have taken exclusive lock, check to see if another thread
+			// recreated the document while we were waiting for the lock
+			response, etag = rdm.cachedResponse, rdm.cachedResponseETag
+			if response == nil {
+				// Re-order the apiGroups by their insertion order
```

Comment:
May run into trouble with this when using this type inside the aggregator.

I have a vague memory there is an ordering stored somewhere. I cant find it now, but will have to look again.

Reply:
Is the comment is a TODO or something that was done? The code before was just looping through the map and adding the entries that way.

Reply:
I am not sure I was reviewing based off of changes you added in dd70b5b51851447c542875b5e0ff0e61a41e3cbc

I initially did have code like this in my original handler but I think a comment from deads caused me to remove it